src/ServiceControl.Audit.AcceptanceTests/Auditing/When_duplicated_messages_are_ingested.cs

```diff
@@ -65,10 +61,14 @@ public AnEndpoint()
 
             class DuplicatingBehavior : Behavior<IAuditContext>
             {
-                public override async Task Invoke(IAuditContext context, Func<Task> next)
+                public override Task Invoke(IAuditContext context, Func<Task> next)
                 {
-                    await next();
-                    await next();
+                    if (context.Message.Headers[Headers.EnclosedMessageTypes].Contains(nameof(DuplicatedMessage)))
+                    {
+                        // create a few duplicate audit messages
+                        return Task.WhenAll(next(), next(), next(), next());
```

Comment:
Technically, this MIGHT cause issues due to concurrent access on shared structures on the context bags which could not occur in normal execution? I can't think of a specific scenario where this would be a problem though. So not against it.